app/models/katello/rhel_lifecycle_status.rb

```diff
@@ -0,0 +1,175 @@
+      },
+      'RHEL7 (ARM)' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2020-11-30')
+      },
+      'RHEL7 (POWER9)' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2021-05-31')
+      },
+      'RHEL6' => {
+        'full_support' => Date.parse('2016-05-10'),
+        'maintenance_support' => Date.parse('2020-11-30'),
+        'extended_support' => Date.parse('2024-06-30')
+      },
+      'RHEL5' => {
+        'full_support' => Date.parse('2009-03-31'),
+        'maintenance_support' => Date.parse('2012-11-30'),
+        'extended_support' => Date.parse('2017-11-30')
+      }
+    }.freeze
+
+    EOS_WARNING_THRESHOLD = 1.year
+
+    def self.status_map
+      map = {
+        full_support: FULL_SUPPORT,
+        maintenance_support: MAINTENANCE_SUPPORT,
+        extended_support: EXTENDED_SUPPORT,
+        approaching_end_of_support: APPROACHING_END_OF_SUPPORT,
+        support_ended: SUPPORT_ENDED
+      }
+
+      map.default = UNKNOWN
+      map
+    end
+
+    def self.to_status(operatingsystem: nil)
+      return UNKNOWN unless operatingsystem.is_a?(::Operatingsystem)
+      release = operatingsystem.rhel_eos_schedule
+      approach_date = warn_date(eos_schedule: release)
+      end_of_support_date = eos_date(eos_schedule: release)
+      return UNKNOWN unless release
+      return UNKNOWN if RHEL_EOS_SCHEDULES[release].nil?
+      return FULL_SUPPORT if Date.today <= RHEL_EOS_SCHEDULES[release]['full_support']
+      return MAINTENANCE_SUPPORT if Date.today <= RHEL_EOS_SCHEDULES[release]['maintenance_support']
+      if approach_date.present? && Date.today >= approach_date && Date.today <= end_of_support_date
+        return APPROACHING_END_OF_SUPPORT
+      end
+      return EXTENDED_SUPPORT if Date.today <= end_of_support_date
+      return SUPPORT_ENDED
+    end
+
+    def self.status_name
+      N_('RHEL lifecycle')
+    end
+
+    def self.humanized_name
+      'rhel_lifecycle'
+    end
+
+    def self.full_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('full_support')
+    end
+
+    def self.maintenance_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('maintenance_support')
+    end
+
+    def self.extended_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('extended_support')
+    end
+
+    def self.eos_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('extended_support') ||
+        RHEL_EOS_SCHEDULES[eos_schedule]&.[]('maintenance_support')
+    end
+
+    def self.warn_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      end_of_support_date = eos_date(eos_schedule: eos_schedule)
```

Comment:
I know RHEL 7 is the one should get the alert this round.
But the full support end date for RHEL 8 is May 31, 2024.
We could display the message when any one of the 3 end of support dates is approaching without too much efforts. What do you think?
` end_of_support =  RHEL_EOS_SCHEDULES[eos_schedule].select { |k, v| (Time.now.utc .. Time.now.utc + EOS_WARNING_THRESHOLD).cover?(v) }`

Reply:
Do we want to alert users when a system is transitioning from full support to maintenance support, etc?  I'm not sure I see how that would add value?

The only time a user actually needs to take action (update to a new OS major) is when approaching the end of support date, unless I'm not understanding something.

Reply:
Alright. Red Hat releases usually follow a ten-year life cycle. Then it is possible to pay for Extended Life Cycle Support (ELS).  Guess that is why we want the alert for RHEL 7 whose maintenance support ends  on June 30, 2024.
Looks like we just need to display the alert message for approaching maintenance support end date or ELS end date.

I don't see the eos approaching message for RHEL 7 yet.
![image](https://github.com/Katello/katello/assets/2593270/c8a4b6aa-1a0b-4868-8347-78319d4e7205)

Reply:
Okay, I can add a `maintenance_warn_date` to go alongside `eos_warn_date`. It will warn 1 year before, is that good?

One thing I wasn't sure about: A couple of the RHEL7 schedules have a "Maintenance support" end date, but no "Extended support" end date. I assume that means that _all_ support for those RHEL7 architectures ends at the Maintenance support end date. Is that correct?

Reply:
> Okay, I can add a `maintenance_warn_date` to go alongside `eos_warn_date`. It will warn 1 year before, is that good?

👍🏻 

> * Support ended (error)

Not sure this one should be an error or warning.

> One thing I wasn't sure about: A couple of the RHEL7 schedules have a "Maintenance support" end date, but no "Extended support" end date. I assume that means that _all_ support for those RHEL7 architectures ends at the Maintenance support end date. Is that correct?

I would agree with you.

Reply:
> Not sure this one should be an error or warning.

If "Support ended" is only a warning, there won't be any differentiation between "Approaching end of support" and "Support ended." For this reason I'd vote to keep it an error status.

(To be clear, error status just means it shows red on host statuses; it doesn't actually throw any errors.) Thoughts?

Reply:
Error status is fine with me. Let's start with that.

Reply:
One more question: Is a host only covered by extended support if the customer has purchased extended support on it?  Currently we have no way to know if extended support has been purchased, so our host statuses must be based on the schedule dates only.

Reply:
I think this is fine. We are just displaying the RHEL lifecyce status not their subscriptions.